resources/test-runner.mjs

```diff
@@ -0,0 +1,96 @@
+import { TEST_INVOKER_LOOKUP } from "./test-invoker.mjs";
+
+export class TestRunner {
+    constructor(suite, test, params, callback, page, frame) {
+        this._suite = suite;
+        this._test = test;
+        this._params = params;
+        this._callback = callback;
+
+        this._page = page;
+        this._frame = frame;
+    }
+
+    async runTest() {
+        // Prepare all mark labels outside the measuring loop.
+        const suiteName = this._suite.name;
+        const testName = this._test.name;
+        const syncStartLabel = `${suiteName}.${testName}-start`;
+        const syncEndLabel = `${suiteName}.${testName}-sync-end`;
+        const asyncStartLabel = `${suiteName}.${testName}-async-start`;
+        const asyncEndLabel = `${suiteName}.${testName}-async-end`;
+
+        let syncTime;
+        let asyncStartTime;
+        let asyncTime;
+        const runSync = () => {
+            this._runWarmupSuite();
+            syncTime = this._measureSyncTime(syncStartLabel, syncEndLabel);
```

Comment:
This code is harder to follow with nested function calls. We should go back to having all the code here instead.

Reply:
I was trying to find a good way to be able to reuse as much as possible for remote workloads. 
If I keep everything in one function, I'll end up overriding the whole function, when I might just need to change one line in one nested function call. 

With this, I should be able to create a new class that extends TestRunner, but only overrides one method, which is more targeted and introduces less CLs.

Reply:
Which is the line that needs to be changed for remote workloads?

Reply:
Some examples, that are slightly different for remote workloads:

`this._test.run(this._page); `
The remote workloads don't use a page object to wrap elements. 
Also, if we opt into async workloads, we might want to await the run function. 

`_forceLayout()`
It works, although I won't use or pass in a frame reference.

These are just some examples

Reply:
Both examples can work as is (even if page and frame are not passed in), but I still feel that smaller methods might be easier to follow and reuse. 

This shouldn't be a blocker though and if I'm the only one feeling this way, I can certainly change it.

Reply:
Perhaps we should make it more of a copy/paste for this PR and then follow up with refactors? I think it'll be easier to say one way or the other with (a) the change unbundled from the file move and (b) when we have a remote implementation to consider.

Reply:
Yeah, doing the split as a follow up seems like a good idea here.

Reply:
sounds good - i'll fix it up!

Reply:
done